src/engine.ml

```diff
@@ -938,10 +939,12 @@ let server_send_push_reply config is_not_taken tls session key tls_data =
     (* XXX(reynir): here we assume the binding for [Key_derivation] is only ever [`Tls_ekm] *)
     @ (if Config.mem Key_derivation config then [ "key-derivation tls-ekm" ]
        else [])
-    @ (Option.to_list (Config.find Protocol_flags config)
-      |> List.map (fun flags ->
-             Fmt.str "%a" (Config.pp_b ?sep:None)
-               (Config.Conf_map.B (Protocol_flags, flags))))
+    @ (Config.find Protocol_flags config
+      |> Option.map (fun flags ->
+             Fmt.str "%a" (Config.pp_b ~sep)
+               (Config.Conf_map.B (Protocol_flags, flags)))
+      |> Option.to_list)
+    @ Option.value ~default:[] (Config.find Push config)
```

Comment:
It's not so nice that this will create e.g. `"push foo,push bar"` if we have `Push, ["foo", "bar"]`

Reply:
does this matter? should we revise in here (should be easy)?

Reply:
No, it doesn't matter much.

Reply:
hmm, actually I don't quite understand. `Push` is a string list. What does the other side expect - is it the string of the values "foo, bar" or "push foo,push bar"?

As I read the code here, if we have "push foo" "push bar" in the configuration (which will turn into `Push ["foo", "bar"]`, we will transmit "foo, bar" over the wire.

Reply:
a `push foo` and `push bar` tells OpenVPN to push `foo` and `bar`. In other words the push reply will look like `PUSH_REPLY,..,foo,bar,..\000` (or any permutation thereof; I don't think the order is important)